userspace/libsinsp/cri.hpp

```diff
@@ -746,33 +737,27 @@ inline bool cri_interface<api>::parse(const libsinsp::cgroup_limits::cgroup_limi
 		}
 	}
 
-	if(!resp.has_status())
+	if(!container_status_resp.has_status())
 	{
-		g_logger.format(sinsp_logger::SEV_DEBUG, "cri (%s) no status, returning", container.m_id.c_str());
+		g_logger.format(sinsp_logger::SEV_DEBUG, "cri (%s): ContainerStatusResponse call no status, returning", container.m_id.c_str());
 		ASSERT(false);
 		return false;
 	}
 
-	const auto &resp_container = resp.status();
-	const auto &resp_container_info = resp.info();
-	container.m_full_id = resp_container.id();
-	container.m_name = resp_container.metadata().name();
-
-	// This is in Nanoseconds(in CRI API). Need to convert it to seconds.
-	container.m_created_time = static_cast<int64_t>(resp_container.created_at() / ONE_SECOND_IN_NS);
-
-	for(const auto &pair : resp_container.labels())
-	{
-		if(pair.second.length() <= sinsp_container_info::m_container_label_max_length)
-		{
-			container.m_labels[pair.first] = pair.second;
-		}
-	}
-
-	parse_cri_image(resp_container, resp_container_info, container);
+	const auto &resp_container = container_status_resp.status();
+	const auto &resp_container_info = container_status_resp.info();
+	const auto root = get_info_jvalue(resp_container_info);
+	parse_cri_base(resp_container, container);
```

Comment:
No more ad-hoc code here, the main `parse` function consistently calls `parse_*` helpers.